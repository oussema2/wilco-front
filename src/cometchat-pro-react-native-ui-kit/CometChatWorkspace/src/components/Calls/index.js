export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
