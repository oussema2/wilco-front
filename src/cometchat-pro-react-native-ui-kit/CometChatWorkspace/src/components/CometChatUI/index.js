/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
