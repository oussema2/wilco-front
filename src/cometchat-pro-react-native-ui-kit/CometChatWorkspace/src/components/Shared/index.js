export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
