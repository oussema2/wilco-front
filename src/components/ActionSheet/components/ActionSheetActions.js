import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
