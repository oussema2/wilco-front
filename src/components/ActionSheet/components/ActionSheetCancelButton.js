import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
