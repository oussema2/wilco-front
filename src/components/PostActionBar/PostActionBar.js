import React from 'react';
import { observer } from 'mobx-react-lite';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import noop from '../../helpers/noop';
import Post from '../../entities/Post';
import { LikeCount } from '../LikeCount';
import { CommentCount } from '../CommentCount';
import { styles } from './PostActionBar.styles';
import { HorizontalPadding } from '../HorizontalPadding';

const PostActionBar = ( {
	testID, post, numberOfLikes, liked, onLikePressed, onCommentPressed
} ) => (
	<View>
		<HorizontalPadding>
			<View testID={testID} style={styles.actionBar}>
				<LikeCount
					testID="action-bar-like-count"
					count={numberOfLikes}
					liked={liked}
					onPress={onLikePressed}
				/>
				<CommentCount testID="post-comment-count" count={post.numberOfComments} onPress={onCommentPressed} />
			</View>
		</HorizontalPadding>
		<LinearGradient start={{ x: 1, y: 0 }} end={{ x: 1, y: 1 }} colors={[ '#f2f2f2', '#FFF' ]} style={{ height: post.numberOfComments ? 10 : 0 }} />
	</View>
);

PostActionBar.propTypes = {
	testID: PropTypes.string,
	post: PropTypes.instanceOf( Post ).isRequired,
	numberOfLikes: PropTypes.number,
	liked: PropTypes.bool,
	onLikePressed: PropTypes.func,
	onCommentPressed: PropTypes.func
};

PostActionBar.defaultProps = {
	testID: 'postActionBar-component',
	numberOfLikes: 0,
	liked: false,
	onLikePressed: noop,
	onCommentPressed: noop
};

export default observer( PostActionBar );
