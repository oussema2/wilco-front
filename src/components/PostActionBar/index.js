export { default as PostActionBar } from './PostActionBar';
