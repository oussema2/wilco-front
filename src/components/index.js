export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
