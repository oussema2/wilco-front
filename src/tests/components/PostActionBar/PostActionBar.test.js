import * as React from 'react';
import {
	render, cleanup, fireEvent
} from '@testing-library/react-native';
import { PostActionBar } from '../../../components/PostActionBar';
import PostFactory from '../../factories/PostFactory';

afterEach( cleanup );

describe( 'PostActionBar', () => {
	const post = PostFactory.build( { numberOfLikes: 4, numberOfComments: 25 } );

	describe( 'with the default props', () => {
		it( 'renders the PostActionBar component correctly', () => {
			const component = render( <PostActionBar post={post} /> );
			expect( component.queryByTestId( 'postActionBar-component' ) ).toBeDefined();
			expect( component.queryByTestId( 'action-bar-like-count' ) ).toHaveTextContent( '0 likes' );
			expect( component.queryByTestId( 'post-comment-count' ) ).toHaveTextContent( '25 comments' );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with the numberOfLikes prop', () => {
		it( 'renders the inner like count with the given number of likes', () => {
			const component = render( <PostActionBar post={post} numberOfLikes={5} /> );
			expect( component.queryByTestId( 'action-bar-like-count' ) ).toHaveTextContent( '5 likes' );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with the liked prop', () => {
		it( 'renders the inner like count with color', () => {
			const component = render( <PostActionBar post={post} liked /> );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with the onLikePressed callback', () => {
		describe( 'when the inner like count is pressed', () => {
			it( 'executes the given callback', () => {
				const onLikePressedMock = jest.fn();
				const { queryByTestId } = render(
					<PostActionBar post={post} onLikePressed={onLikePressedMock} />
				);
				fireEvent.press( queryByTestId( 'action-bar-like-count' ) );
				expect( onLikePressedMock ).toHaveBeenCalledTimes( 1 );
			} );
		} );
	} );
} );
