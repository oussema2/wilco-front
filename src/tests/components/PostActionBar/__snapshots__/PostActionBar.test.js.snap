// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostActionBar with the default props renders the PostActionBar component correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 20,
        },
        null,
      ]
    }
    testID="horizontalPadding-component"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 24,
          "paddingBottom": 20,
        }
      }
      testID="postActionBar-component"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="action-bar-like-count"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/thumbsUp.png",
              }
            }
            style={
              Object {
                "height": 16.67,
                "tintColor": "#5B5B5B",
                "width": 16.67,
              }
            }
            testID="like-count-image"
          />
          <Text
            style={
              Object {
                "color": "#5B5B5B",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 13,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 9.67,
              }
            }
            testID="like-count-text"
          >
            0 likes
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="post-comment-count"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/comment.png",
              }
            }
            style={
              Object {
                "height": 14.58,
                "tintColor": "#5B5B5B",
                "width": 16.67,
              }
            }
            testID="comment-count-image"
          />
          <Text
            style={
              Object {
                "color": "#5B5B5B",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 13,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 9.67,
              }
            }
            testID="comment-count-text"
          >
            25 comments
          </Text>
        </View>
      </View>
    </View>
  </View>
  <BVLinearGradient
    colors={
      Array [
        4294111986,
        4294967295,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    style={
      Object {
        "height": 10,
      }
    }
  />
</View>
`;

exports[`PostActionBar with the liked prop renders the inner like count with color 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 20,
        },
        null,
      ]
    }
    testID="horizontalPadding-component"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 24,
          "paddingBottom": 20,
        }
      }
      testID="postActionBar-component"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="action-bar-like-count"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/thumbsUp.png",
              }
            }
            style={
              Object {
                "height": 16.67,
                "tintColor": "#12A3BE",
                "width": 16.67,
              }
            }
            testID="like-count-image"
          />
          <Text
            style={
              Object {
                "color": "#12A3BE",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 13,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 9.67,
              }
            }
            testID="like-count-text"
          >
            0 likes
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="post-comment-count"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/comment.png",
              }
            }
            style={
              Object {
                "height": 14.58,
                "tintColor": "#5B5B5B",
                "width": 16.67,
              }
            }
            testID="comment-count-image"
          />
          <Text
            style={
              Object {
                "color": "#5B5B5B",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 13,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 9.67,
              }
            }
            testID="comment-count-text"
          >
            25 comments
          </Text>
        </View>
      </View>
    </View>
  </View>
  <BVLinearGradient
    colors={
      Array [
        4294111986,
        4294967295,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    style={
      Object {
        "height": 10,
      }
    }
  />
</View>
`;

exports[`PostActionBar with the numberOfLikes prop renders the inner like count with the given number of likes 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 20,
        },
        null,
      ]
    }
    testID="horizontalPadding-component"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 24,
          "paddingBottom": 20,
        }
      }
      testID="postActionBar-component"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="action-bar-like-count"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/thumbsUp.png",
              }
            }
            style={
              Object {
                "height": 16.67,
                "tintColor": "#5B5B5B",
                "width": 16.67,
              }
            }
            testID="like-count-image"
          />
          <Text
            style={
              Object {
                "color": "#5B5B5B",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 13,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 9.67,
              }
            }
            testID="like-count-text"
          >
            5 likes
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="post-comment-count"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/comment.png",
              }
            }
            style={
              Object {
                "height": 14.58,
                "tintColor": "#5B5B5B",
                "width": 16.67,
              }
            }
            testID="comment-count-image"
          />
          <Text
            style={
              Object {
                "color": "#5B5B5B",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 13,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 9.67,
              }
            }
            testID="comment-count-text"
          >
            25 comments
          </Text>
        </View>
      </View>
    </View>
  </View>
  <BVLinearGradient
    colors={
      Array [
        4294111986,
        4294967295,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    style={
      Object {
        "height": 10,
      }
    }
  />
</View>
`;
